pkg/component/component.go

```diff
@@ -239,3 +243,83 @@ func Exists(client *occlient.Client, componentName, applicationName, projectName
 	}
 	return false, nil
 }
+
+// LinkInfo contains the information about the link being created between
+// components
+type LinkInfo struct {
+	SourceComponent string
+	TargetComponent string
+	Envs            []string
+}
+
+// Link injects connection information of the target component into the source
+// component as environment variables
+func Link(client *occlient.Client, sourceComponent, targetComponent, applicationName string) (*LinkInfo, error) {
+	// Get Service associated with the target component
+	targetService, err := client.GetOneServiceFromSelector(
+		util.ConvertLabelsToSelector(
+			componentlabels.GetLabels(
+				targetComponent, applicationName, false,
+			),
+		),
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get service associated with the component %v", targetComponent)
+	}
+
+	// Generate environment variables to inject
+	linkHostKey := fmt.Sprintf("COMPONENT_%v_HOST", strings.ToUpper(targetComponent))
+	linkHostValue := targetService.Name
+
+	linkPortKey := fmt.Sprintf("COMPONENT_%v_PORT", strings.ToUpper(targetComponent))
+	linkPort, err := getPortFromService(targetService)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get port from Service %v", targetService.Name)
+	}
+	linkPortValue := fmt.Sprint(linkPort)
+
+	// Inject environment variable to source component
+	sourceDC, err := client.GetOneDeploymentConfigFromSelector(
+		util.ConvertLabelsToSelector(
+			componentlabels.GetLabels(
+				sourceComponent, applicationName, false,
+			),
+		),
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Deployment Config for component %v", sourceComponent)
+	}
+
+	if err := client.AddEnvironmentVariablesToDeploymentConfig(
+		[]corev1.EnvVar{
+			{
+				Name:  linkHostKey,
+				Value: linkHostValue,
+			},
+			{
+				Name:  linkPortKey,
+				Value: linkPortValue,
+			},
+		}, sourceDC); err != nil {
+		return nil, errors.Wrapf(err, "unable to add environment variables to Deployment Config %v", sourceDC.Name)
+	}
+
+	return &LinkInfo{
+		SourceComponent: sourceComponent,
+		TargetComponent: targetComponent,
+		Envs: []string{
+			fmt.Sprintf("%v=%v", linkHostKey, linkHostValue),
+			fmt.Sprintf("%v=%v", linkPortKey, linkPortValue),
+		},
+	}, nil
+}
+
+// getPortFromService returns the first port listed in the service
+func getPortFromService(service *corev1.Service) (int32, error) {
```

Comment:
Do we really need a separate function that is just one if statement and nothing more? This is used just in one place.

Reply:
imo it makes the Link() function look cleaner

Reply:
IMO it makes it harder to read ;-)